kernel/src/log_segment/mod.rs

```diff
@@ -0,0 +1,374 @@
+//! Represents a segment of a delta log. [`LogSegment`] wraps a set of  checkpoint and commit
+//! files.
+
+use crate::actions::{get_log_schema, Metadata, Protocol, METADATA_NAME, PROTOCOL_NAME};
+use crate::path::ParsedLogPath;
+use crate::schema::SchemaRef;
+use crate::snapshot::CheckpointMetadata;
+use crate::utils::require;
+use crate::{
+    DeltaResult, Engine, EngineData, Error, Expression, ExpressionRef, FileSystemClient, Version,
+};
+use itertools::Itertools;
+use std::cmp::Ordering;
+use std::convert::identity;
+use std::sync::{Arc, LazyLock};
+use tracing::warn;
+use url::Url;
+
+#[cfg(test)]
+mod tests;
+
+/// A [`LogSegment`] represents a contiguous section of the log and is made of checkpoint files
+/// and commit files and guarantees the following:
+///     1. Commit file versions will not have any gaps between them.
+///     2. If checkpoint(s) is/are present in the range, only commits with versions greater than the most
+///        recent checkpoint version are retained. There will not be a gap between the checkpoint
+///        version and the first commit version.
+///     3. All checkpoint_parts must belong to the same checkpoint version, and must form a complete
+///        version. Multi-part checkpoints must have all their parts.
+///
+/// [`LogSegment`] is used in [`Snapshot`] when built with [`LogSegment::for_snapshot`], and
+/// and in `TableChanges` when built with [`LogSegment::for_table_changes`].
+///
+/// [`Snapshot`]: crate::snapshot::Snapshot
+#[derive(Debug)]
+#[cfg_attr(feature = "developer-visibility", visibility::make(pub))]
+pub(crate) struct LogSegment {
+    pub end_version: Version,
+    pub log_root: Url,
+    /// Sorted commit files in the log segment (ascending)
+    pub sorted_commit_files: Vec<ParsedLogPath>,
```

Comment:
Rescuing https://github.com/delta-incubator/delta-kernel-rs/pull/495#discussion_r1843195141
> qq: Would it be more self-describing to call this ascending_commit_files?

Reply:
I can rename it to that 👍